patterns/multisig/VirtualMultisigRune.ts

```diff
@@ -0,0 +1,244 @@
+import { type Event as ProxyEvent } from "polkadot_dev/types/pallet_proxy/pallet.ts"
+import { type RuntimeEvent } from "polkadot_dev/types/polkadot_runtime.ts"
+import {
+  Chain,
+  ChainRune,
+  Event,
+  ExtrinsicSender,
+  hex,
+  MetaRune,
+  MultiAddress,
+  Rune,
+  RunicArgs,
+  ValueRune,
+} from "../../mod.ts"
+import { $ } from "../../mod.ts"
+import { replaceDelegateCalls } from "../proxy/mod.ts"
+import { Multisig, MultisigRune } from "./MultisigRune.ts"
+
+export interface VirtualMultisig extends Multisig {
+  stash: Uint8Array
+}
+export const $virtualMultisig: $.Codec<VirtualMultisig> = $.object(
+  $.field("signatories", $.array($.sizedUint8Array(32))),
+  $.optionalField("threshold", $.u8),
+  $.field("stash", $.sizedUint8Array(32)),
+)
+
+export class VirtualMultisigRune<out U, out C extends Chain = Chain>
+  extends Rune<VirtualMultisig, U>
+{
+  inner
+  signatories
+  threshold
+  stash
+  encoded
+  hex
+
+  constructor(_prime: VirtualMultisigRune<U>["_prime"], readonly chain: ChainRune<U, C>) {
+    super(_prime)
+    this.inner = this.into(MultisigRune, chain)
+    const v = this.into(ValueRune)
+    this.signatories = v.access("signatories")
+    this.threshold = v.access("threshold")
+    this.stash = v.access("stash")
+    this.encoded = v.map((m) => $virtualMultisig.encode(m))
+    this.hex = this.encoded.map(hex.encode)
+  }
+
+  ratify<X>(...[senderIndex, call]: RunicArgs<X, [number, Chain.Call<C>]>) {
+    const call_ = this.chain.extrinsic(Rune.rec({
```

Comment:
Proxy call needs to happen outside of ratify. We can't assume every ratify call will interact with the stash